### ISSUE TITLE
Question - Accessibility Standard supported by MISP (EN 301 549 / WCAG / ARIA)

### ISSUE THREAD

**Issue by mdowens** [MISP/MISP#4258]
Could anybody please tell me if MISP supports any User Interface Accessibility standard? 
e.g. EN 301 549  or  WCAG v2

I can see many references to ARIA in the UI code but cannot see any documented reference if any Accessibility Standards are supported.

Thanks for your help in advance
Regards, Mike

**Comment by adulau** [comment]
Hi Mike,

We received some contributions and made some improvements to the accessibility aspect in MISP in the past years. Not really sure if the latest code is really compatible with ARIA, there are maybe some potential issues which need to be fixed. I hope this helps. 

Cheers 

PS: If you spot any specific issues, let us know.  Thank you

**Comment by mdowens** [comment]
Hi Alexandre,
Thanks for your message.  My customer is a UK government agency which requires its IT systems to be compliant with accessibility standards.  We have conducted an "Accessibility Test" against MISP 2.4.101 (latest version at the time) and have found a number of issues ( 5 issues that would prevent usage of MISP and 2 issues that would impair usage of MISP ).  The accessibility technologies used for the test were "ZoomText JAWS 2019 Screen Reader, ZoomText 2019 Screen Reader & Dragon Naturally Speaking". 
I have pasted in a detailed report but in summary the issues found were:
1. Page: Administration, List Roles - unreadable table information
2. Page: Event Actions, List Events  then  View an Event, Sightings  (span id=”sightingCount_nnnnnnn”) -cannot read sightings value.
3. Page: Event Actions, Import from - pop-up not accessible
4. Page: View Event, + Correlation Graph - cannot access any correlation graph content
5. Many Pages – “Distribution” - Distribution Graph cannot be opened using a keyboard
6. Page: Event Actions, View Events - Attribute Search Filter input field does not have a text label
    	      Attribute display panel input field does not have a text label
7. Page: Global Actions, Statistics - “Activity Heatmap” Combo Box does not have a Text Label.

I hope this information is helpful to you.
Best regards, Mike
	 
Accessibility Impairment Testing Products Used:
ZoomText Fusion 2019	
JAWS 2019 Screen Reader
ZoomText 2019 Screen Magnification & Screen Reader

Dragon Naturally Speaking   Voice activated software for users with a motor impairment
	

Tested against	MISP Version	2.4.101


Severity of Issue:
Severity Level 1 - Customers with a disability cannot use the product efficiently and are unable to work around the problem.  The product forces customers to behave in a way which causes them great difficulty.  Customers with a disability are unable to perceive what they should do in common circumstances

Severity Level 2 - The product functions in way that customers with a disability would find time consuming, unsatisfactory and confusing.  Where there is a work around for customers with a disability it could have a detrimental effect on user satisfaction, efficiency and effectiveness.  Customers with a disability cannot respond effectively to an event i.e. takes too long, or product causes users to make errors when doing so


Severity Level 1 Issues Found:
1.	Page:	Administration, List Roles
Unreadable table information within the Roles screen. 	 Affected - JAWS 
Impact: It was not possible to establish which options within this table were selected. The ticks used are not textual data that can be read by a screen reader.


2.	Page: Event Actions, List Events  then  View an Event, Sightings  (span id=”sightingCount_nnnnnnn”)  		
Impact: It was possible to change the sightings counter using the buttons but was not possible to read the current sightings value  (not textual).		Affected - JAWS
 


3.	Page: Event Actions, Import from     (“Choose the format that you would like to use for the import – MISP standard, STIX 1.1.1, STIX 2.0”)
Impact: Pop-up not accessible – user not notified of the pop-up and the pop-up form cannot be read by a Screen Reader (JAWS, Dragon, ZoomText)  	Affected - JAWS, Dragon, ZoomText

 

4.	Page: View Event, + Correlation Graph		
Impact: Unable to access Correlation Graph panel content with any assistive technology.			Affected - JAWS, ZoomText, Dragon


5.	Many Pages – “Distribution” 
Impact: Distribution Graph can only be opened using a mouse (cannot be opened using the keyboard) 			Affected -  JAWS, Dragon, ZoomText


Severity Level 2 Issues Found:
6.	Page: Event Actions, View Events
Impact: Attribute Search Filter input field (input id=”quickFilterField”) does not have a text label		Affected - WCAG 1.1.1, JAWS, Dragon, ZoomText
Impact: Attribute display panel input field (id=”PostMessage”) does not have a text label			Affected - WCAG 1.1.1, JAWS, Dragon, ZoomText
		

7.	Page: Global Actions, Statistics
Impact: “Activity Heatmap” Combo Box does not have a Text Label.	 	
Affected - WCAG 1.1.1, JAWS, Dragon, ZoomText

**Comment by iglocska** [comment]
Hello Mike,

excellent summary, thanks a lot, we'll try to fix the identified issues, for internal use a checklist:

- [x] Page: Administration, List Roles - unreadable table information
- [ ] Page: Event Actions, List Events then View an Event, Sightings (span id=”sightingCount_nnnnnnn”) -cannot read sightings value.
- [ ] Page: Event Actions, Import from - pop-up not accessible
- [ ] Page: View Event, + Correlation Graph - cannot access any correlation graph content
- [ ] Many Pages – “Distribution” - Distribution Graph cannot be opened using a keyboard
- [ ] Page: Event Actions, View Events - Attribute Search Filter input field does not have a text label
Attribute display panel input field does not have a text label
- [ ] Page: Global Actions, Statistics - “Activity Heatmap” Combo Box does not have a Text Label.

**Comment by mdowens** [comment]
Hi Andreas,

Thanks very much for your quick reply. 

I have seen another accessibility question on the support forum a while ago and I really appreciate how seriously the MISP development community regard this stuff. 

My customer is the UK tax agency and we are also going through a full infrastructure and application penetration test of our AWS hosted MISP server. Are you interested in the results of this? (Emailed to you directly obviously). 

Thanks for making MISP such a great system. 

Regards, Mike

> On 29 Mar 2019, at 20:02, Andras Iklody <notifications@github.com> wrote:
> 
> Hello Mike,
> 
> excellent summary, thanks a lot, we'll try to fix the identified issues, for internal use a checklist:
> 
>  Page: Administration, List Roles - unreadable table information
>  Page: Event Actions, List Events then View an Event, Sightings (span id=”sightingCount_nnnnnnn”) -cannot read sightings value.
>  Page: Event Actions, Import from - pop-up not accessible
>  Page: View Event, + Correlation Graph - cannot access any correlation graph content
>  Many Pages – “Distribution” - Distribution Graph cannot be opened using a keyboard
>  Page: Event Actions, View Events - Attribute Search Filter input field does not have a text label
>  Attribute display panel input field does not have a text label
>  Page: Global Actions, Statistics - “Activity Heatmap” Combo Box does not have a Text Label.
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
--------------------------------------------------------------------------------------------------
Marlborough Solutions Ltd, 17 Burnside Gardens, Walsall WS53LB
Registered Office: 1 Crossways Court, Haslemere Road, Fernhurst  GU27 3EP
Registered in England  No. 3534202

The information in this email is confidential and may be legally privileged.  
It is intended solely for the addressee. Access to this email by anyone-else is 
unauthorised.  If you are not the intended recipient, any disclosure, copying, 
distribution or any action taken or omitted to be taken in reliance on it, is 
prohibited and may be unlawful. When addressed to our clients any opinions or 
advice contained in this email are subject to the terms and conditions 
expressed in any applicable governing Marlborough Solutions Ltd terms of 
business.

**Comment by iglocska** [comment]
Hello there, 

definitely interested! If you could send it pgp encrypted to info@circl.lu using http://pgp.circl.lu/pks/lookup?op=get&search=0xEAADCFFC22BD4CD5 that would be great!

Thanks a lot in advance.

**Comment by obert01** [comment]
Hello all,

Working in France in the Cert-IST, I am visually impaired (nearly blind) and I am daily using MISP with the JAWS or NVDA screen readers (this clearly depends on my mood ;-) ). 

I have already got a few bug fixed, and contributed a few pull requests on that topic. I am ready to provide support on that topic if needed. 

In general, the most problematic accessibility issues in MISP can be fixed by setting the "aria-label" and "role" properties on empty "span" items. 

Since recently, there are also many unlabelled links of the form:
```
<a href="xxx" class="fa fa-xxx" title="blablabla"></a>
```
IMPORTANT: The "title" property is not sufficient to make an HTML element accessible. In fact, the "title" is only presented to the user when the element is focused or hovered. So, this is not sufficient as a blind user will generally use a virtual review mode of his/her screen reader, which does not focuses items as the page is reviewed. 

I have a few modifications to propose on this particular aspect, but they are not finalized. I will submit a pull request when ready.

**Comment by iglocska** [comment]
Awesome @obert01, as always it's much appreciated. Will keep the title vs aria label issue in mind!

**Comment by djsherly** [comment]
Hi All,

is there an update available on WCAG 2.1 compliance?

**Comment by ZerhackerJagi** [comment]
Hey there!
I was wondering if there’s any progress on this thread. Does the team have a roadmap or a general document regarding accessibility for MISP? It would be a great signal to see compliance with standards such as ISO 8241 171, EN 301 549, WCAG.
Thanks in advance.

**Comment by adulau** [comment]
There is also a support the ARIA labels in MISP and we are talking into consideration this for the new UI (3.0).

We recently add a new part concerning audio assistant in MISP to provide audio support for describing event.

https://discourse.ossbase.org/t/audio-assistant-in-misp/1060

We are also evaluating to use it for the graph visualisation. 

Concerning the ISO standards, we would be more than happy to have a copy of those standards for review.
